src/ReadWriteLocks.jl

```diff
@@ -49,7 +39,7 @@ write_lock(rwlock::ReadWriteLock) = rwlock.write_lock
 
 function lock!(read_lock::ReadLock)
```

Comment:
This API should be changed to match the `AbstractLock` API, including the other API functions like `trylock`.

Reply:
You mean also rename `lock` -> `lock!` (even though we mutate the `rwlock.readers`)?

https://github.com/invenia/ReadWriteLocks.jl/issues/6